src/Breweries.php

```diff
@@ -4,20 +4,38 @@
 
 namespace Interns2022B;
 
+use Illuminate\Support\Collection;
+use Interns2022B\Models\Brewery;
+
 class Breweries
 {
     public const SEARCH = "choice";
 
-    public function getBreweries(array $breweriesData): array
+    public function getBreweries(Collection $breweriesFactory): Collection
     {
         $name = readline("Please provide brewery or city name:");
 
-        $chosenData = [];
-        foreach ($breweriesData as $rowValue) {
-            if (in_array($name, $rowValue, strict: true)) {
-                $chosenData[] = $rowValue;
+        $chosenData = new Collection();
+        $chosenData->push(["Brewery name", "City", "Country", "Provider"]);
+
+        $breweriesFactory->each(function (Brewery $item) use ($chosenData, $name): void {
+            if ($item->city->name === $name || $item->name === $name) {
+                $chosenData->push([$item->name, $item->city->name, $this->getCountry($item->countries), $this->getProvider($item->providers)]);
             }
-        }
-        return $chosenData;
+        });
+
+        return ($chosenData->count() > 1)
+            ? $chosenData
+            : new Collection([["Record doesn't exist"]]);
+    }
+
+    protected function getCountry(Collection $countries): string
+    {
+        return $countries->first();
+    }
+
+    protected function getProvider(Collection $providers): string
+    {
+        return $providers->first();
```

Comment:
Why first country and provider? What if there will be more than 1 item? Maybe lets use `implode()` method on Collection object
https://laravel.com/docs/9.x/collections#method-implode

Reply:
As I noticed, it is currently impossible in our code to get more than one provider from `.json` file
Initially it looks like `BreweryFactory.php` lets us to collect more than one provider in `Collection`, but every time we take `"provider"` through `BreweryFactory.php`, it makes us new Collection with only one record inside it.
That's why we use `first()`. This solution is temporary used, until we will collect real data from websites.

Reply:
Got it